cmd/cli/plugin/package/go.mod

```diff
@@ -0,0 +1,248 @@
+module github.com/vmware-tanzu/tanzu-framework/cmd/cli/plugin/package
+
+go 1.17
+
+replace (
+	github.com/vmware-tanzu/tanzu-framework => ../../../../
```

Comment:
Why do we need tanzu-framework and apis/run depedencies?

Reply:
> Why do we need tanzu-framework...

I think until PR #3260 is merged, I'll have a dependency on TF. I use it in `cmd/cli/plugin/package/main.go` for the `github.com/vmware-tanzu/tanzu-framework/pkg/v1/sdk/capabilities/discovery` package.

```go
apiGroup1 := capdiscovery.Group("packageMetadateAPIQuery", packagedatamodel.DataPackagingAPIName).WithVersions(packagedatamodel.PackagingAPIVersion).WithResource("packagemetadatas")

...
	return clusterQueryClient.Query(apiGroup1, apiGroup2, apiGroup3, apiGroup4).Execute()
```

And as we discussed, I still have a TF dependency because of `cmd/cli/plugin/package/test/package_plugin_integration_test.go`. I'll remove that dependency in a separate PR, after the following dependencies get moved out of the TF module:
```
"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/test/framework"
"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/test/framework/exec"
```

> ...and apis/run depedencies?

The above-mentioned `discovery` package depends on `runv1alpha1 "github.com/vmware-tanzu/tanzu-framework/apis/run/v1alpha1"` in `pkg/v1/sdk/capabilities/discovery/generate.go`, and so I depend on it as well.